src/test/java/bio/terra/service/filedata/DrsServiceTest.java

```diff
@@ -651,14 +667,91 @@ public void postAccessUrlForObjectIdInvalidPassport() {
         UnauthorizedException.class,
         () ->
             drsService.postAccessUrlForObjectId(
-                googleDrsObjectId, "gcp-passport-us-central1", drsPassportRequestModel));
+                googleDrsObjectId, "gcp-passport-us-central1", drsPassportRequestModel, null));
+  }
+
+  @Test
+  public void testSignGcpUrlNoProject() {
+    testSignGcpUrl(null, false);
+  }
+
+  @Test
+  public void testSignGcpUrlSpecifiedProject() {
+    testSignGcpUrl("myProject", false);
+  }
+
+  @Test
+  public void testSignGcpUrlSpecifiedProjectSelfHosted() {
+    testSignGcpUrl("myProject", true);
+  }
+
+  public void testSignGcpUrl(String billingProject, boolean selfHostedDataset) {
+    SnapshotSummaryModel snapshotSummary =
+        new SnapshotSummaryModel().id(snapshotId).selfHosted(selfHostedDataset);
+    Snapshot snapshot =
+        mockSnapshot(snapshotId, billingProfile.getId(), CloudPlatform.GCP, SNAPSHOT_DATA_PROJECT);
+    snapshot.getSourceDataset().getDatasetSummary().selfHosted(selfHostedDataset);
+
+    if (selfHostedDataset) {
+      String sourcePath = "gs://path/to/file.txt";
+      String datasetProject = snapshot.getSourceDataset().getProjectResource().getGoogleProjectId();
+      snapshotSummary.selfHosted(selfHostedDataset);
+      Storage storage = mock(Storage.class);
+      Bucket bucket = mock(Bucket.class);
+      when(bucket.getLocation()).thenReturn("us-central1");
+      when(storage.get(eq(BlobId.fromGsUtilUri(sourcePath).getBucket()), any())).thenReturn(bucket);
+      // This mock is needed in order to get the region from the bucket
+      when(gcsProjectFactory.getStorage(datasetProject)).thenReturn(storage);
+
+      when(samService.signUrlForBlob(
+              eq(TEST_USER), eq(billingProject), eq(sourcePath), eq(URL_TTL)))
+          .thenReturn(
+              "https://storage.googleapis.com/path/to/file.txt"
+                  + "?requestedBy=%s&userProject=%s"
+                      .formatted(
+                          URLEncoder.encode(TEST_USER.getEmail(), StandardCharsets.UTF_8),
+                          billingProject));
+    }
+    when(snapshotService.retrieve(snapshotId)).thenReturn(snapshot);
+    when(snapshotService.retrieveSnapshotSummary(snapshotId)).thenReturn(snapshotSummary);
+
+    DRSObject object = drsService.lookupObjectByDrsId(TEST_USER, googleDrsObjectId, false);
+    DRSAccessMethod accessMethod = object.getAccessMethods().get(0);
+    assertThat(
+        "Only BEARER authorization is included",
+        accessMethod.getAuthorizations().getSupportedTypes().size(),
+        equalTo(1));
```

Comment:
You could use the `hasSize` matcher instead:

```
accessMethod.getAuthorizations().getSupportedTypes(), hasSize(1)
```

But on second glance, is the assertion overly broad? Can we instead check that it's the BEARER auth included specifically like the message indicates?